script.sh

```diff
@@ -85,9 +88,46 @@ else
   BUNDLE_EXEC="bundle exec "
 fi
 
+if [ "${INPUT_ONLY_CHANGED}" = "true" ]; then
+  echo '::group:: Getting changed files list'
+
+  # check if commit is present in repository, otherwise fetch it
+  if ! git cat-file -e "${BASE_REF}"; then
+    git fetch --depth 1 origin "${BASE_REF}"
+  fi
+
+  # get intersection of changed files (excluding deleted) with target files for
+  # rubocop as an array
+  # shellcheck disable=SC2086
+  readarray -t CHANGED_FILES < <(
+    comm -12 \
+      <(git diff --diff-filter=d --name-only "${BASE_REF}..${HEAD_REF}" | sort || kill $$) \
+      <(${BUNDLE_EXEC}rubocop --list-target-files | sort || kill $$)
+  )
+
+  if (( ${#CHANGED_FILES[@]} == 0 )); then
+    echo "No relevant files for rubocop, skipping"
+    exit 0
+  fi
+
+  printf '%s\n' "${CHANGED_FILES[@]}"
+
+  if (( ${#CHANGED_FILES[@]} > 100 )); then
+    echo "More than 100 changed files (${#CHANGED_FILES[@]}), running rubocop on all files"
+    unset CHANGED_FILES
+  fi
+
+  echo '::endgroup::'
+fi
+
 echo '::group:: Running rubocop with reviewdog 🐶 ...'
 # shellcheck disable=SC2086
-${BUNDLE_EXEC}rubocop ${INPUT_RUBOCOP_FLAGS} --require ${GITHUB_ACTION_PATH}/rdjson_formatter/rdjson_formatter.rb --format RdjsonFormatter \
+${BUNDLE_EXEC}rubocop \
+  ${INPUT_RUBOCOP_FLAGS} \
+  --require ${GITHUB_ACTION_PATH}/rdjson_formatter/rdjson_formatter.rb \
+  --format RdjsonFormatter \
+  --fail-level error \
```

Comment:
@toy 
Did you want to intentionally change `fail-level`?
I think it seems a bug.

Reply:
@kehra It was intentional, due to turning on pipefail, which in turn is needed for pipes in getting the list of changed files.
Before this commit the exit code of `rubocop … | reviewdog …` was the exit code of reviewdog. But with pipefail and without changing `fail-level`, any rubocop violation would make the script have an error exit code, which will be confusing, as both violations will be reported and the action will fail.

Reply:
@toy 
Thank you for your response.
I think this change caused our CI to miss non-error level(e.g. warning) violations.
However, my colleague has made it possible to specify the fail-level in #120, so it is not a problem 😄

Reply:
@kehra #120 looks good, it allows to "shoot in the foot" by changing formatter, but I think it is better to restrict less.

I'm still wondering about your CI - any violations of any level will create either annotation or comment to PR, so the rubocop outcome will be failing, but not the rubocop action itself, am I missing something?

Reply:
@toy 
Hmm.
I might be wrong, so I'll investigate this again 💦 
Sorry for bothering you.

Reply:
No worries, I'm just trying to understand if there is some case that allows action to not report violations. Please reply with your findings